### PR TITLE
Fix server bugs with patch

### PR DESCRIPTION
**Pull request by knoxy5467**

### CODE REVIEW

File: common/src/lib.rs [hunk 1]
```diff
@@ -122,6 +126,17 @@ pub struct TaskProp {
     pub value: TaskPropVariant,
 }
 
+/// fields of tasks wihtout any properties
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub enum PrimitiveField {
```
Comment (knoxy5467) [inline common/src/lib.rs:131]: good adding primitives for task parts

File: common/src/lib.rs [hunk 2]
```diff
@@ -135,6 +150,15 @@ pub enum Filter {
         /// Immediate value to use with the comparator
         immediate: TaskPropVariant,
     },
+    /// property of the task itself
+    LeafPrimitive {
```
Comment (knoxy5467) [inline common/src/lib.rs:154]: good, proper names

**Review by knoxy5467 — COMMENTED** [review]
looks good to me

**Review by jacobe90 — COMMENTED** [review]
Still a lot of weird behavior with the app which is concerning; newly created tasks don't save after quitting and restarting. I assume this will all be fixed when we merge zon-dev into main?

**Pull request merged**